packages/venia-concept/src/components/MiniCart/product.js

```diff
@@ -152,8 +154,18 @@ class Product extends Component {
 
     removeItem = () => {
         // TODO: prompt user to confirm this action
-        this.props.removeItemFromCart({
-            item: this.props.item
+
+        const { removeItemFromCart, loadingElement, item } = this.props;
+
+        this.setState({
+            isOpen: false
+        });
+
+        loadingElement && loadingElement(true);
+
+        removeItemFromCart({
+            item: item,
+            loadingElement: loadingElement
```

Comment:
Passing this function feels odd to me. Not only is it named vaguely (`isLoadingElement` would be better) but it doesn't seem like something that needs to be passed to the `removeItemFromCart` action.

If `removeItemFromCart` is async you could `await` the value and then call `isLoadingElement(false)` afterwards instead of having to pass it through:

```js
isLoadingElement(true)
await removeItemFromCart({...})
isLoadingElement(false)
```

Reply:
Outdated conversation.